public/scripts/project/lock.js

```diff
@@ -0,0 +1,75 @@
+define(function(require) {
+  var fs = Bramble.getFileSystem();
+  var sh = new fs.Shell();
+  var LOCK_KEY = 'bramble-lock';
+  var EXPIRY = 2000; // MS
+
+  function lock(root, callback) {
+    function refresh(callback) {
+      fs.setxattr(root, LOCK_KEY, Date.now(), function(err) {
```

Comment:
You don't need this function body (i.e., you're not doing anything special): `fs.setxattr(root, LOCK_KEY, Date.now(), callback);`